basicrta/gibbs.py

```diff
@@ -227,6 +236,17 @@ def cluster(self, method="GaussianMixture", **kwargs):
         :param method: Mixture method to use
         :type method: str
         """
+        # Check if this Gibbs result was created from combined contact data
+        if hasattr(self, '_from_combined_contacts') and self._from_combined_contacts:
+            raise NotImplementedError(
+                "Kinetic clustering is not yet supported for combined contact data. "
+                "The trajectory source information needed for proper kinetic clustering "
+                "is available in the combined contact files but not yet utilized in the "
+                "clustering algorithm. For now, analyze each trajectory separately for "
+                "kinetic clustering, or use the combined data only for residence time "
+                "distribution analysis."
+            )
+            
```

Comment:
This function does NOT perform kinetic clustering (also know as "kinetic mapping"). The kinetic mapping functionality lives in kinetics.KineticMap. 

Ensure that cluster() works with combined data.